test/ArcxAnalyticsSdk.test.ts

```diff
@@ -1,70 +1,84 @@
-import { SdkConfig } from '../src/types'
-import { MetaMaskInpageProvider } from '@metamask/providers'
 import sinon from 'sinon'
 import { expect } from 'chai'
+import { ArcxAnalyticsSdk, SdkConfig } from '../src'
 import {
+  ATTRIBUTION_EVENT,
+  CAUGHT_TRANSACTION_EVENT,
+  CHAIN_CHANGED_EVENT,
   CONNECT_EVENT,
-  PAGE_EVENT,
-  TRANSACTION_EVENT,
-  DEFAULT_SDK_CONFIG,
   CURRENT_URL_KEY,
-  REFERRER_EVENT,
-  FIRST_PAGE_VISIT,
+  DEFAULT_SDK_CONFIG,
   DISCONNECT_EVENT,
-  CHAIN_CHANGED_EVENT,
-  CAUGHT_TRANSACTION_EVENT,
+  FIRST_PAGE_VISIT,
+  IDENTITY_KEY,
+  PAGE_EVENT,
+  REFERRER_EVENT,
+  TRANSACTION_EVENT,
 } from '../src/constants'
 import * as postRequestModule from '../src/helpers/postRequest'
 import {
+  TEST_ACCOUNT,
+  TEST_API_KEY,
+  TEST_CHAIN_ID,
+  TEST_IDENTITY,
   TEST_JSDOM_URL,
   TEST_REFERRER,
   TEST_UTM_CAMPAIGN,
   TEST_UTM_MEDIUM,
   TEST_UTM_SOURCE,
-} from './jsdom.setup.test'
-import { TEST_ADDRESS, TEST_CHAIN_ID, TEST_IDENTITY } from './fixture'
+} from './fixture'
+import { MetaMaskInpageProvider } from '@metamask/providers'
 import { MockEthereum } from './MockEthereum'
-import { ArcxAnalyticsSdk } from '../src'
-
-const PROD_URL_BACKEND = DEFAULT_SDK_CONFIG.url // Backwards compatability
 
 const ALL_FALSE_CONFIG: SdkConfig = {
+  ...DEFAULT_SDK_CONFIG,
   cacheIdentity: false,
   trackPages: false,
   trackReferrer: false,
   trackUTM: false,
   trackWalletConnections: false,
   trackChainChanges: false,
   trackTransactions: false,
-  url: PROD_URL_BACKEND,
-}
-const TEST_API_KEY = '01234'
-const TEST_ATTRIBUTES = {
-  a: 'value',
-  b: 'second value',
 }
 
 describe('(unit) ArcxAnalyticsSdk', () => {
-  describe('Base functionality', () => {
-    let postRequestStub: sinon.SinonStub
-    let analyticsSdk: ArcxAnalyticsSdk
-
-    beforeEach(async () => {
-      postRequestStub = sinon.stub(postRequestModule, 'postRequest').resolves(TEST_IDENTITY)
-      analyticsSdk = await ArcxAnalyticsSdk.init(TEST_API_KEY, ALL_FALSE_CONFIG)
-      postRequestStub.resetHistory()
-      sessionStorage.clear()
-    })
+  let analyticsSdk: ArcxAnalyticsSdk
+  let postRequestStub: sinon.SinonStub
+
+  beforeEach(async () => {
+    const ethereumMock = sinon.createStubInstance(MetaMaskInpageProvider)
+    window.ethereum = ethereumMock
+    postRequestStub = sinon.stub(postRequestModule, 'postRequest').resolves(TEST_IDENTITY)
+
+    analyticsSdk = await ArcxAnalyticsSdk.init(TEST_API_KEY, ALL_FALSE_CONFIG)
+
+    postRequestStub.resetHistory()
+    localStorage.clear()
+    sessionStorage.clear()
+  })
 
-    afterEach(() => sinon.restore())
+  afterEach(sinon.restore)
 
+  describe('Public Functions', () => {
     describe('#init', () => {
```

Comment:
Initialisation should be moved outside of 'Public Functions'. In my opinion it is like contstructor, not Public function. Also what is the point of splitting functions into `Private functions` and `Public Functions` if you introduced naming convention and we already have difference in naming

Reply:
refactored